Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs

```diff
@@ -214,6 +217,43 @@ public float TeleportDuration
             set { teleportDuration = value; }
         }
 
+        [SerializeField]
+        [Tooltip("Profile for wiring up diagnostic assets.")]
+        private MixedRealityDiagnosticsProfile diagnosticsSystemProfile;
+
+        /// <summary>
+        /// Active profile for diagnostic configuration
+        /// </summary>
+        public MixedRealityDiagnosticsProfile DiagnosticsSystemProfile
+        {
+            get { return diagnosticsSystemProfile; }
+            private set { diagnosticsSystemProfile = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("Enable diagnostic system")]
+        private bool enableDiagnosticsSystem = false;
+
+        public bool IsDiagnosticsSystemEnabled
+        {
+            get { return enableDiagnosticsSystem && DiagnosticsSystemSystemType?.Type != null; }
+            private set { enableDiagnosticsSystem = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("Diagnostics System Class to instantiate at runtime.")]
+        [Implements(typeof(IMixedRealityDiagnosticsManager), TypeGrouping.ByNamespaceFlat)]
```

Comment:
We should also probably add this to the IsCoreManager check in the MixedRealityManager class